tasks/kvm_host/install-storage.yml

```diff
@@ -0,0 +1,150 @@
+  file:
+    path: /root/.ssh/storage_rsa.pub
+    mode: '0600'
+
+- name: Hard-code root pwd for the storage VM
+  set_fact:
+    storage_root_pwd: "/root/.ssh/storage_rsa.pub"
+
+- name: Hard-code root pwd for the storage VM
+  set_fact:
+    vm_name: "storage-0"
+
+############## Copied from old RHUI playbook, LIKELY need to need some changes #######
+  # Optional
+- name: Create VM root disk (based on the original Fedora os qcow2 image, so that each storage node(s)' disk(s) file will just store the diff, and not store the whole OS once more)
+  command: "qemu-img create -f qcow2 -F qcow2 -b {{ storage_disk_path }} {{guests_image_base_path}}{{item.key}}.qcow2"
+  with_dict: "{{storage}}"
+  become: true
+
+# references: https://access.redhat.com/documentation/en-us/red_hat_update_infrastructure/3.1/html-single/system_administrators_guide/index#gather_required_information
+- name: Expand root disk to the correct size
+  command: "qemu-img resize  {{ copied_storage_root_pwd }} +{{spec_storage.disk}}G"
+  with_items: "{{ storage }}"
+  become: true
+
+- name: Workaround to expand the disks, part I
+  command: " cp {{guests_image_base_path}}{{item.key}}.qcow2 {{guests_image_base_path}}{{item.key}}.qcow2.orig"
+  with_dict: "{{storage}}"
+  become: true 
+
+# virt-resize can NOT expand in-place
+# references: https://linux.die.net/man/1/virt-resize
+- name: Workaround to expand the disks, part II (this will take some time)
+  shell: "virt-resize --expand /dev/vda1 {{guests_image_base_path}}{{item.key}}.qcow2.orig {{guests_image_base_path}}{{item.key}}.qcow2"
+  with_dict: "{{storage}}"
+  become: true
+
+- name: Confirm disks were expanded correctly
+  shell: "virt-filesystems --long -h --all -a {{guests_image_base_path}}{{item.key}}.qcow2 | grep partition | awk '{ gsub(\"G\",\"\",$6); print $6}'"
+  with_dict: "{{storage}}"
+  become: true
+  register: expanded_disks
+
+# FIX me: fedora 38 will have a different magic number!
+# NOTE: 7.8G is a magic number. i.e. it is the initial disk size of the latest RHEL7.9 qcow2 image, as per Aug 2021
+# references: https://access.redhat.com/downloads/content/69/ver=/rhel---7/7.9/x86_64/product-software
+- name: Fail if the disks were not expanded correctly
+  fail:
+    msg: "Disk for {{item.item.key}} ws not resized properly"
+  when:
+    - item.stdout == "7.8"
+    - "'hypervisor' in group_names"
+  with_list: "{{expanded_disks.results}}"
+
+- name: Delete VMs disks used to expand the root disk, since the new disk has expanded correctly
+  file:
+    path: "{{guests_image_base_path}}{{item.key}}.qcow2.orig"
+    state: absent
+  with_dict: "{{storage}}"
+  ignore_errors: true
+############## Copied from old RHUI playbook ##############
+
+
+# want fix: move to using the virt module...
+# Throws following error on RHEL8.9 <===============
+# Fix the following error before removing WIP from the PR <========
+ # "virt-customize: error: libguestfs error: inspect_os: mount exited with ", <========
```

Comment:
This is expected: 
[From RHEL8 docs](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/considerations_in_adopting_rhel_8/file-systems-and-storage_considerations-in-adopting-rhel-8)

Reply:
Fixed by switching to Fedora Server